internal/updateFunction/main.go

```diff
@@ -0,0 +1,52 @@
+package update
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cloudevents/sdk-go/v2/event"
+	"github.com/sirupsen/logrus"
+	firestore "pantheon.io/edrt-policy-docs-functions/internal/store"
+	"pantheon.io/edrt-policy-docs-functions/internal/types"
+)
+
+type UpdateHandler struct {
+	logger *logrus.Logger
+	store  *firestore.Firestore
+}
+
+func NewUpdateHandler(logger *logrus.Logger, fs *firestore.Firestore) UpdateHandler {
+	return UpdateHandler{
+		logger: logger,
+		store:  fs,
+	}
+}
+
+func (u *UpdateHandler) PolicyDocUpdated(ctx context.Context, e event.Event) error {
+	/**
+	* This function is intended to be the pub/sub interface. It will do as little as
+	* possible. It will call into other packages to read from firestore, denormalize
+	* the data, then write to firestore
+	**/
+
+	logger := u.logger.WithField("Function", "PolicyDocUpdated")
+
+	var msg types.MessagePublishedData
+
+	if err := e.DataAs(&msg); err != nil {
+		return fmt.Errorf("event.DataAs: %v", err)
+	}
+
+	msgText := string(msg.Message.Data)
+
+	logger.WithField("PubSub", true).WithField("Data", msgText).Info("Pubsub triggered")
+	logger.Info("Iterating over attributes")
+	for x := range msg.Message.Attributes {
+		logger.WithField(x, msg.Message.Attributes[x]).Info("Attribute")
+	}
+	logger.Info("Iteration complete")
+
+	result := u.store.Read(ctx)
```

Comment:
Maybe we can introduce the service layer here and keep the read/transform/write logic inside. Otherwise, it looks like we are mixing the transport/presentation layer from PubSub (reading bytes, decoding, casting the message input data, etc.) with the persistent layer here (reading straight from the Firestore). 

Side note here. As I remember, for cloud functions we will need to deploy all code we have instead of the single binary. Hence, we will deploy all the source code (even other cloud function we have in this repo).  Means having a dedicated service layer will be ok, and we do not need to have a subpackage for every possible cloud function.

Reply:
> Maybe we can introduce the service layer here and keep the read/transform/write logic inside

Yes, excellent idea, I'll do this

> even other cloud function we have in this repo

This goes back to what I mentioned in my original post in this PR. What do you think? I wrote this in a way that this repo would contain multiple cloud functions, but I can also see us having one repository per cloud function. Do you have a preference? I think @mattcallahan had thoughts on this too.